### PR TITLE
Results wrapper

### PR DESCRIPTION
**Pull request by albfan**
Relates to #259 

The only thing I dislike is to keep short name for deprecated method: e.g `getTimeEntries()` vs `getTimeEntriesResultsWrapper()`. Commits like 682e8a614006753ae809455baa56f1c1d6ff3f6e simply remove old methods `getAssignee()` with new ones `getAssigneeName()` and `getAssigneeId()` but I want to ask before propose a break in compatibility.

Even more I don't know if a single commit is preferred.

I can rebase and squash after code review

### CODE REVIEW

File: src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.taskadapter.redmineapi.internal;
+
+import java.util.List;
+
+public class ResultsWrapper<T> {
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java]: let's make this final

File: src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    final private Integer totalFoundOnServer;
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java]: oracle/sun conventions suggest different order of modifiers: "private final"

File: src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    public ResultsWrapper(Integer totalFoundOnServer, Integer limitOnServer, Integer offsetOnServer, List<T> results) {
+        this.totalFoundOnServer = totalFoundOnServer;
+        this.limitOnServer = limitOnServer;
+        this.offsetOnServer = offsetOnServer;
+        this.results = results;
+    }
+
+    public boolean hasSomeResults() {
+        return !results.isEmpty();
```
Comment (alexeyOnGitHub) [inline src/main/java/com/taskadapter/redmineapi/internal/ResultsWrapper.java]: I assume the list is never null. right?

**Comment by alexeyOnGitHub** [comment]
thank you for submitting this.  I also appreciate that you take backward compatibility into consideration.
let's drop the old methods since this will go to the new major version - 3.0.0 and the compatibility is less of a concern now.

when all done, please squash into 1 commit.

**Comment by alexeyOnGitHub** [comment]
note - I extracted ResultsWrapper to a separate file in another commit just now.

**Comment by albfan** [comment]
I have done all fixes, and squashed it. I will fix the conflict with ResultWrapper rebasing it. One second

**Comment by albfan** [comment]
Done

**Comment by alexeyOnGitHub** [comment]
super! thank you, @albfan

**Pull request merged**